### ISSUE TITLE
Photos: broken, no photos displayed; avatars: broken, no avatars displayed

### ISSUE THREAD

**Issue by AlfredSK** [friendica/friendica#8103]
### Expected behavior
When I go to `friendicadomain.tld/photos/$user` I should see my uploaded photos in my albums.
User/Contact avatars should show the user's profile picture.

### Actual behavior
When I go to `friendicadomain.tld/photos/$user` I see no photos at all.
User/Contact avatars are ~~partially~~ missing. A placeholder for 'image not found' is shown instead. ~~Approx. 50% of the avatar images are missing/broken.~~

Edit: No. It's not partially. The images that show up are cached in the browser's cache. When I clean the cache all avatars are missing as well.

No PHP errors are thrown.

I guess this is related to:
https://github.com/friendica/friendica/issues/7998#issuecomment-573179114 and...
https://github.com/friendica/friendica/issues/7998#issuecomment-573290305

### Steps to reproduce the problem
**I use 'filesystem' as storage** driver and 'redis' for session and caching if that's important.
I guess this is caused by one of the refactoring commits that were merged lately. Maybe @nupplaphil has an idea.

### Friendica version you encountered the problem
2020.03-dev

### Friendica source (git, zip)
git

### PHP version
7.2

### SQL version
10.1 MariaDB

**Comment by AlfredSK** [comment]
Browser console is showing a lot of errors when images are loaded:

```
GET https://mordor.social/photo/94816067915bc12469ddefe720809211-6.jpg?ts=1578624182
[HTTP/2 500 Internal Server Error 89ms]

```
or

```
GET https://mordor.social/photo/34991067315de3fd0d95eff982080406-2.jpg
[HTTP/2 500 Internal Server Error 86ms]

```
or

```
GET https://mordor.social/photo/68956032025b0a9d86f1f2b359484126-6.jpg?ts=1575491382
[HTTP/2 500 Internal Server Error 80ms]


```

**Comment by AlfredSK** [comment]
Ok. I checked out commit ee58a5589 Now it is working again. So, this is caused by the changes to the storage or cache stuff.

**Comment by MrPetovan** [comment]
Paging @nupplaphil

**Comment by copiis** [comment]
I confirm

**Comment by MrPetovan** [comment]
You should get PHP error messages along with the 500 errors.

**Comment by AlfredSK** [comment]
> You should get PHP error messages along with the 500 errors.

Nope.

**Comment by tobiasd** [comment]
I don't get no errors as well, the one with the OpenStreetmal addon was the only one. The images are simply all missing, as long as they should be loaded from the cache it seems.

**Comment by nupplaphil** [comment]
I'm sorry that I'm so silent the last days, but I got ill... I think I can hopefully check/fix this issue at the end of the week.. Sorry for the inconvenience :/

**Comment by AlfredSK** [comment]
@nupplaphil Don't worry about it. Get well! 🍵 😷 🚑

**Comment by tobiasd** [comment]
Get well soon @nupplaphil and don't worry!

**Comment by AlfredSK** [comment]
Just an addition. We have several nodes on develop but only a few seem to have this issue. So I think it probably depends on the cache driver the node is using. I'm using redis. 
Would be interesting what the others are using.

**Comment by nupplaphil** [comment]
Now I'd need the result of the command, I recently had posted:
```sql
MariaDB [friendica]> select distinct `backend-class` from photo;
```

which should result only textual entries, **no** classes (`\Friendica\Model\Storage\...`) - my result e.g. is:
```sql
+---------------+
| backend-class |
+---------------+
| Filesystem    |
|               |
| NULL          |
+---------------+
3 rows in set (2.723 sec)
```

If there are other entries, this indicate a wrong migration, which could cause such troubles

**Comment by nupplaphil** [comment]
hm .. just found some minor bugs yet, but nothing which should cause SUCH troubles, it really seems like a wrong - or in progress - migration of these nodes. Is that possible? @tobiasd @AlfredSK plz  paste both the result of 
```sql
select distinct `backend-class` from attach;
select distinct `backend-class` from photo;
```

thx ..

**Comment by AlfredSK** [comment]
Ok.

```
MariaDB [friendica]> select distinct `backend-class` from photo;
+------------------------------------+
| backend-class                      |
+------------------------------------+
| Friendica\Model\Storage\Filesystem |
+------------------------------------+
1 row in set (8.15 sec)
```
and

```
MariaDB [friendica]> select distinct `backend-class` from attach;
+------------------------------------+
| backend-class                      |
+------------------------------------+
| Friendica\Model\Storage\Filesystem |
+------------------------------------+
1 row in set (0.00 sec)
```
Looks wrong. :-)

**Comment by AlfredSK** [comment]
And a different result with another develop node:

```
MariaDB [mordor]> select distinct `backend-class` from photo;
+------------------------------------+
| backend-class                      |
+------------------------------------+
| Friendica\Model\Storage\Filesystem |
|                                    |
| Filesystem                         |
+------------------------------------+
3 rows in set (2.71 sec)
```
and a node on master:
```
MariaDB [venera]> select distinct `backend-class` from photo;
+------------------------------------+
| backend-class                      |
+------------------------------------+
| Friendica\Model\Storage\Filesystem |
+------------------------------------+
1 row in set (1.66 sec)

```

So, on every node I operate there are class paths not simple textual entries.

**Comment by nupplaphil** [comment]
hmmm.. We found the source!! On `develop` branches, it should be `Filesystem` while on `master` branches, it should stay at `Friendica\Model\Storage\Filesystem`... 

https://mordor.social/friendica says you're operating on DB version 1329. The update script was for 1330. Is there maybe an update running, or was there an error during the update?

**Comment by AlfredSK** [comment]
I ran the dbstructure update and the postupdate. The latter took 1.5 hours on mordor and several hours on libranet.
On both nodes I'm not on current develop anymore. I checked out commit commit ee58a55 to make the nodes work again.

**Comment by AlfredSK** [comment]
When I checkout develop it says...
`Die Datenbankversion ist 1330 und die Post-Update-Version 1329.`
dbstructure update and postupdate exit w/o any error - "All pending post updates are done."

**Comment by AlfredSK** [comment]
Btw. is it normal that backend-class is an empty string for some 'photo' rows?

**Comment by AlfredSK** [comment]
I updated both tables manually...

``` 
MariaDB [mordor]> update `attach` set `backend-class` = 'Filesystem' where `backend-class` like '%Filesystem%';
Query OK, 1 row affected (0.00 sec)
Rows matched: 1  Changed: 1  Warnings: 0

MariaDB [mordor]> select distinct `backend-class` from attach;
+---------------+
| backend-class |
+---------------+
| Filesystem    |
+---------------+
1 row in set (0.00 sec)

MariaDB [mordor]> update `photo` set `backend-class` = 'Filesystem' where `backend-class` like '%Filesystem%';
Query OK, 325363 rows affected (3.12 sec)
Rows matched: 325537  Changed: 325363  Warnings: 0

MariaDB [mordor]> select distinct `backend-class` from photo;
+---------------+
| backend-class |
+---------------+
| Filesystem    |
|               |
+---------------+
2 rows in set (1.08 sec)
``` 
Now it is working on current develop. Could it be that the db update script was stumbling over the backslashes in the backend-class entries?

**Comment by nupplaphil** [comment]
> I ran the dbstructure update and the postupdate. The latter took 1.5 hours on mordor and several hours on libranet.
> On both nodes I'm not on current develop anymore. I checked out commit [ee58a55](https://github.com/friendica/friendica/commit/ee58a55895cf3c7b44f40c8ac466143845e753f9) to make the nodes work again.

OK, that's why you're now mixed .. After the postupdate, there "shouldn't" be any class-names left.. But after you switched back to the commit, it saves again with the class-names :-) ..

**Comment by nupplaphil** [comment]
> Btw. is it normal that backend-class is an empty string for some 'photo' rows?

A mystery, which was before the recent update too.. I didn't dive into this issue yet :-)

**Comment by nupplaphil** [comment]
> Now it is working on current develop. Could it be that the db update script was stumbling over the backslashes in the backend-class entries?

Looks good now! Yes, maybe the backslashes are the problem .. I'll check it :-)

**Comment by AlfredSK** [comment]
> OK, that's why you're now mixed .. After the postupdate, there "shouldn't" be any class-names left.. But after you switched back to the commit, it saves again with the class-names :-) ..

Yeah. I ran mordor on develop (with the updated DB) for several days before I noticed the problem.

**Comment by AlfredSK** [comment]
Oh! I found something interesting for the empty backend-class entries. There were 540 rows with backend-class empty and backend-ref empty.

So I ran...

```
sudo -u www-data php bin/console.php storage move photo
[2020-01-17 21:54:30] Moved 540 files
[2020-01-17 21:54:30] Moved 540 files total
```
and now it looks like this...
```
MariaDB [mordor]> select `backend-class`, `backend-ref` from `photo` where `backend-class` LIKE '';
Empty set (0.18 sec)

MariaDB [mordor]> select distinct `backend-class` from photo;
+---------------+
| backend-class |
+---------------+
| Filesystem    |
+---------------+
1 row in set (0.95 sec)
```
Purrrfect. :-)

**Comment by nupplaphil** [comment]
thx for THIS hint ;-) .. I'll try it on my node too

**Comment by AlfredSK** [comment]
Btw. I found a bug for `bin/console.php storage move`. Without a second argument move should move 'attach' and 'photo' but it fails with...
`Argument 2 passed to Friendica\Core\StorageManager::move() must be of the type array, null given, called in /var/www/html/mordor/src/Console/Storage.php on line 150`

**Comment by nupplaphil** [comment]
> Btw. I found a bug for `bin/console.php storage move`. Without a second argument move should move 'attach' and 'photo' but it fails with...
> `Argument 2 passed to Friendica\Core\StorageManager::move() must be of the type array, null given, called in /var/www/html/mordor/src/Console/Storage.php on line 150`

thx & fixed with #8123 too

**Comment by AlfredSK** [comment]
Oh! There is another weird thing. :-)

```
sudo -u www-data php bin/console.php storage list
 Sel | Name                
-----------------------
     | Filesystem          
     | Database            

The current storage class (Filesystem) is not registered!
```
What does the last line mean? There should be an asterisk (*) in front of 'Filesystem'. On master it looks like this...
```
sudo -u www-data php bin/console.php storage list          
 Sel | Name                
-----------------------
 *   | Filesystem          
     | Database
```
